src/app/account/repo/add/form.js

```diff
@@ -16,8 +17,22 @@ const initialState = {
 
 export default function Form({ usage }) {
   const [state, formAction] = useFormState(getRepo, initialState);
-  const disabled = usage >= process.env.NEXT_PUBLIC_REPO_LIMIT ? true : false;
+  const [usageLimitReached,setUsageLimitReached] = useState(false)
+  const [githubUrl , setGithubUrl] = useState(state?.data?.url || '')
+  const [isGithubUrlValid,setIsGithubUrlValid] = useState(false)
+  const [isDisabled,setIsDisabled] = useState(true)
 
+  const githubRegex = /^https:\/\/github\.com\/[a-zA-Z0-9_-]+\/[a-zA-Z0-9_-]+$/;
```

Comment:
My concern with this, is we should let github return a `404` if the repo is not found - if we try to do it with a regex we might be stopping people adding repos

Reply:
After more thought, I am concerned that this might present other bugs in the future

Reply:
> My concern with this, is we should let github return a `404` if the repo is not found - if we try to do it with a regex we might be stopping people adding repos

I suggested regex so people can't enter non repo links. Like git lab links or spam links, or anything else

Reply:
I think the UX is better, but could create a bug without us knowing. At the moment we rely on github returning a `200` or `404` which after some thought I think is better

Reply:
okay so I think we could only check for the domain using regex i.e. whatever the URL is being entered has to be a GitHub URL and let's not check further as if the repo does not exist the GitHub itself will return a 404.

Example: 
https://google.com/EddieHubCommunity/HealthCheck --> is a invalid URL
https://github.com/EddieHubCommunity/Reop --> returns a 404 from github
https://github.com/EddieHubCommunity/HealthCheck --> a valid URL

Reply:
great idea, that is a great idea and comprise 👍